In [1]:
import json
import os

import pandas as pd
import numpy as np

from dotenv import load_dotenv

from core_ds4a_project import cleaning, columns as project_columns, datasets

%load_ext autoreload
%autoreload 1
%aimport core_ds4a_project, core_ds4a_project.cleaning, core_ds4a_project.columns, core_ds4a_project.datasets

pd.set_option("display.max_columns", None)

Environment variables:

In [2]:
load_dotenv('envvars')

ROOT_DATA_PATH = os.environ.get('ROOT_DATA_PATH')
RAW_DATA_PATH = os.environ.get('RAW_DATA_PATH') or f'{ROOT_DATA_PATH}/raw'

Reading data:

In [3]:
df = datasets.read_castigada_xlsx(RAW_DATA_PATH, clean=False)

df.shape, df.columns.sort_values()

((10416, 15),
 Index(['CLASE', 'DEUDOR', 'FECHA_CASTIGO', 'INT_RECUPERADOS', 'K_NO_MIGRADO',
        'LLAVE_LINIX', 'NIT', 'NO_MIGRADOS_A_LINIX', 'OBLIGACION',
        'OPCION_LLAVE_LINIX', 'PAGARE', 'RECUPERACION_K', 'SALDO_POR_RECUPERAR',
        'SIF', 'VR_CASTIGO'],
       dtype='object'))

Renaming columns:

In [4]:
df.columns = cleaning.normalize_columns_name(df.columns)

df.columns.sort_values()  # sorting for presentational purposes only

Index(['CLASE', 'DEUDOR', 'FECHA_CASTIGO', 'INT_RECUPERADOS', 'K_NO_MIGRADO',
       'LLAVE_LINIX', 'NIT', 'NO_MIGRADOS_A_LINIX', 'OBLIGACION',
       'OPCION_LLAVE_LINIX', 'PAGARE', 'RECUPERACION_K', 'SALDO_POR_RECUPERAR',
       'SIF', 'VR_CASTIGO'],
      dtype='object')

Filtering CASTIGO records containing known OBLIGACION (1170 out of 10416):

In [5]:
count_castigo_all = df.shape[0]
df = df.dropna(subset='OBLIGACION')

{
    "count_castigo_all": count_castigo_all,
    "count_castigo_known_obligacion": df.shape[0],
}

{'count_castigo_all': 10416, 'count_castigo_known_obligacion': 1170}

CASTIGO records with known OBLIGACION are dated since 2020-04 only:

In [6]:
df['FECHA_CASTIGO'].min()

Timestamp('2020-04-01 00:00:00')